File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/bridge/transport/Plane.java
package com.bigboxcode.designpattern.bridge.transport;

public class Plane extends Transport {

    public Plane(Seat seat) {
        super(seat);
    }

    @Override
    public void selectTransport() {
        // Write code to select transport
        System.out.println("Plane selected for transport");

        // Select seat
        seat.selectSeat();
    }
}

File: design-patterns/ts/bridge/transport-seat/plane.ts
// plane.ts

import Seat from "./seat";
import Transport from "./transport";

class Plane extends Transport {
    constructor(seat: Seat) {
        super(seat);
    }

    selectTransport(): void {
        console.log("Plane selected for transport");
        this.seat.selectSeat();
    }
}

export default Plane;